processing/src/test/java/io/druid/query/cache/CacheKeyBuilderTest.java

```diff
@@ -286,11 +286,9 @@ public void testNotEqualCacheables()
 
   private static void assertNotEqualsEachOther(List<byte[]> keys)
   {
-    for (byte[] k1 : keys) {
-      for (byte[] k2 : keys) {
-        if (k1 != k2) {
-          assertFalse(Arrays.equals(k1, k2));
-        }
+    for (int i = 0; i < keys.size(); i++) {
+      for (int j = i + 1; j < keys.size(); j++) {
+        assertFalse(Arrays.equals(keys.get(i), keys.get(j)));
```

Comment:
As far as I understand, the previous version was looking for duplicates in the keys list. Condition `k1 != k2` was filtering "the diagonal" in n^2 search, but it was unreliable. Anyway, I had to change this piece of code because inspection about array reference comparison.